esp32-hal/examples/embassy_serial.rs

```diff
@@ -52,25 +52,28 @@ async fn run(mut uart: Uart<'static, UART0>) {
         } else {
             wbuf.clear();
             write!(&mut wbuf, "\r\n-- received {} bytes --\r\n", rbuf.len()).unwrap();
-            embedded_hal_async::serial::Write::write(&mut uart, wbuf.as_slice())
+            embedded_io_async::Write::write(&mut uart, wbuf.as_slice())
                 .await
                 .unwrap();
-            embedded_hal_async::serial::Write::write(&mut uart, rbuf.as_slice())
+            embedded_io_async::Write::write(&mut uart, rbuf.as_slice())
                 .await
                 .unwrap();
-            embedded_hal_async::serial::Write::write(&mut uart, b"\r\n")
+            embedded_io_async::Write::write(&mut uart, b"\r\n")
                 .await
                 .unwrap();
         }
-        embedded_hal_async::serial::Write::flush(&mut uart)
-            .await
-            .unwrap();
+        embedded_io_async::Write::flush(&mut uart).await.unwrap();
 
         // set rbuf full capacity
         rbuf.resize_default(rbuf.capacity()).ok();
         let mut offset = 0;
         loop {
-            match with_timeout(READ_TIMEOUT, uart.read(&mut rbuf[offset..])).await {
+            match with_timeout(
+                READ_TIMEOUT,
+                embedded_io_async::Read::read(&mut uart, &mut rbuf[offset..]),
```

Comment:
Not relevant to this PR but maybe we should remove the eh reexports from out prelude to stop us having to do stuff like this :thinking: